lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -67,22 +181,27 @@ fn sphere_lat(lat: f64) -> f64 {
 }
 
 /// Get neighbour geohash even from the other side of coordinates
-fn sphere_neighbor(hash_str: &str, direction: Direction) -> Result<GeoHash, GeohashError> {
-    let (coord, lon_err, lat_err) = decode(hash_str)?;
+fn sphere_neighbor(hash: GeoHash, direction: Direction) -> Result<GeoHash, GeohashError> {
+    let hash_str = SmolStr::from(hash);
+    let (coord, lon_err, lat_err) = decode(hash_str.as_str())?;
     let (dlat, dlng) = direction.to_tuple();
     let lon = sphere_lon(coord.x + 2f64 * lon_err.abs() * dlng);
     let lat = sphere_lat(coord.y + 2f64 * lat_err.abs() * dlat);
 
     let neighbor_coord = Coord { x: lon, y: lat };
-    encode(neighbor_coord, hash_str.len()).map(Into::into)
+    encode(neighbor_coord, hash_str.len())
+        .map(Into::<SmolStr>::into)
+        .map(Into::into)
```

Comment:
Let's also implement `impl From<String> for GeoHash` so we don't have to convert into an intermediate `SmolStr`

Reply:
At this point we could just re-implement the `encode` method and avoid using strings (especially heap-allocated ones).

It's not that big, and it uses similar `u64` packing internally already.